src/Ganeti/HTools/ExtLoader.hs

```diff
@@ -117,11 +117,15 @@ loadExternalData opts = do
         | setLuxi -> wrapIO . Luxi.loadData $ fromJust lsock
         | setSim -> Simu.loadData simdata
         | setFile -> wrapIO . Text.loadData $ fromJust tfile
-        | setIAllocSrc -> wrapIO . IAlloc.loadData $ fromJust iallocsrc
+        -- IAlloc.loadData calls updateMissing internally because Hail does not
+        -- loadExternalData for loading the JSON config (see wrapReadRequest).
+        -- Here we just pass a 0 as the 'generic' call to updateMissing follows.
+        | setIAllocSrc -> wrapIO . flip IAlloc.loadData 0 $ fromJust iallocsrc
```

Comment:
Because it doesn't really matter what value is specified there. IAlloc has a different interface that doesn't use loadExternalData (see readRequest) that's why IAlloc's loadData does not only loads but also calls mergeData/updateMissing. Here however, mergeData/updateMissing follows a few lines below, so the values calculated by IAlloc's loadData are overwritten. This is the consequence of what I described in the first comment.